From 69b5466f54cc61171afca184b52e9a56ec0104c1 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Tue, 7 Nov 2017 18:40:19 +0800
Subject: [PATCH 2755/5242] MLK-16774 usb: gadget: utp: fix coverity CID
 414727 issue

commit  e7bc1ee1c2cf91f768d518c407edb2fa8e396fa4 from
https://source.codeaurora.org/external/imx/linux-imx.git

CID 414727: Explicit null dereferenced (FORWARD_NULL)
var_deref_op: Dereferencing null pointer uud.

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/gadget/function/fsl_updater.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/gadget/function/fsl_updater.c b/drivers/usb/gadget/function/fsl_updater.c
index 97860fa..521834f 100644
--- a/drivers/usb/gadget/function/fsl_updater.c
+++ b/drivers/usb/gadget/function/fsl_updater.c
@@ -444,6 +444,10 @@ static int utp_exec(struct fsg_dev *fsg,
 		if (uud->data.flags & UTP_FLAG_REPORT_BUSY)
 			pr_info("\tBUSY\n");
 #endif
+	} else {
+		pr_err("UTP write list is empty.\n");
+		mutex_unlock(&ctx->lock);
+		return 0;
 	}
 	mutex_unlock(&ctx->lock);
 
-- 
1.7.9.5

